app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/SearchResultService.cs

```diff
@@ -148,6 +158,39 @@ private async Task<bool> UploadNames(Person person, SSG_SearchRequest request, S
                 return false;
             }
         }
+        private async Task<bool> UploadEmployment(Person person, SSG_SearchRequest request, SSG_Person ssg_person, int? providerDynamicsID, CancellationToken concellationToken)
+        {
+            if (person.Employments == null) return true;
+            try
+            {
+                foreach (var employment in person.Employments)
+                {
+                    EmploymentEntity e = _mapper.Map<EmploymentEntity>(employment);
+                    e.SearchRequest = request;
+                    e.InformationSource = providerDynamicsID;
+                    e.Person = ssg_person;
+                    SSG_Employment ssg_employment  = await _searchRequestService.CreateEmployment(e, concellationToken);
+
+                    //add phones
+
+                    foreach(var phone in employment.Employer.Phones)
+                    {
+                        SSG_PhoneNumber p = _mapper.Map<SSG_PhoneNumber>(phone);
+                        p.LinkedEmployment = ssg_employment;
+                        p.SearchRequest = request;
```

Comment:
shall we link employer phone to search request?

Reply:
I think so, Dynamics would option to display all phones regardless of source for a search request  in a single tab and use the link between phone and employment to determine where it came from.

Reply:
I also do not think we should link this phone to person.
But, as your explanation, it may give Dyanmics more flexibility.and Dynamics is good with this solution. So, I am fine with it.